Mastodon/Scene/MediaPreview/MediaPreviewViewController.swift

```diff
@@ -26,24 +24,39 @@ final class MediaPreviewViewController: UIViewController, NeedsDependency {
         
     let visualEffectView = UIVisualEffectView(effect: UIBlurEffect(style: .systemMaterial))
     let pagingViewController = MediaPreviewPagingViewController()
-    
-    let closeButtonBackground: UIVisualEffectView = {
-        let backgroundView = UIVisualEffectView(effect: UIBlurEffect(style: .systemUltraThinMaterial))
-        backgroundView.alpha = 0.9
-        backgroundView.layer.masksToBounds = true
-        backgroundView.layer.cornerRadius = MediaPreviewViewController.closeButtonSize.width * 0.5
-        return backgroundView
+
+    let topToolbar: UIStackView = {
+        class TouchTransparentStackView: UIStackView {
+            // allow button hit boxes to grow outside of this view’s bounds
+            override func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
+                subviews.contains { $0.point(inside: $0.convert(point, from: self), with: event) }
+            }
+
+            // allow taps on blank areas to pass through
+            override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+                let view = super.hitTest(point, with: event)
+                if view == self {
+                    return nil
+                }
+                return view
+            }
+        }
```

Comment:
Is there a reason not to put this in its own file? Or don't you see the necessity (yet) because if that changes, it's pretty easy to refactor into its own file?

It also feels a bit overengineered, to use a StackView instead of a simple button. Maybe you can say a few words, why you did it this way?

Reply:
I can extract it! The issue is that the area of the stack view between the elements (in this case the entire top of the screen below the notch) is considered part of the stack view and as a result touches don’t pass through down to the view that causes the dismissal of the media preview. (SwiftUI fixes this by default 😉)